packages/lesswrong/components/posts/PostsItem.jsx

```diff
@@ -59,94 +71,189 @@ class PostsItem extends PureComponent {
+        postsViewed,
+        post,
+        // from withMutation, GraphQL HOC
+        increasePostViewCount,
+      } = this.props;
+      // a post id has been found & it's has not been seen yet on this client session
+      if (post && post._id && postsViewed && !postsViewed.includes(post._id)) {
+
+        // trigger the asynchronous mutation with postId as an argument
+        await increasePostViewCount({postId: post._id});
+
+        // once the mutation is done, update the redux store
+        setViewed(post._id);
+      }
+
+      //LESSWRONG: register page-visit event
+      if (this.props.currentUser) {
+        const eventProperties = {
+          userId: this.props.currentUser._id,
+          important: false,
+          intercom: true,
+        };
+
+        eventProperties.documentId = post._id;
+        eventProperties.postTitle = post.title;
+        this.props.registerEvent('post-view', eventProperties)
+      }
+  }
+
+  renderHighlightMenu = () => {
+    return (
+      <div className="posts-item-highlight-footer" >
+        <span className="posts-item-hide-highlight" onTouchTap={this.toggleHighlight}>
+          <FontIcon className={classNames("material-icons")}>
+            subdirectory_arrow_left
+          </FontIcon>
+          Collapse
+        </span>
+        <span className="posts-item-view-full-post">
+          <Link to={Posts.getPageUrl(this.props.post)}>
+            Continue to Full Post
+          </Link>
+        </span>
+      </div>
+    )
+  }
+
   render() {
 
-    const {post, inlineCommentCount} = this.props;
+    const {post, inlineCommentCount, currentUser} = this.props;
     const read = post.lastVisitedAt;
     const newComments = post.lastVisitedAt < post.lastCommentedAt;
+
     const postLink = this.getPostLink()
     const commentCount = post.commentCount ? post.commentCount : 0
 
-
     let postClass = "posts-item";
     if (post.sticky) postClass += " posts-sticky";
 
-    const commentCountIconStyle = {
-      width: '30px',
-      height: '30px',
-      color: read ? (
-        newComments ? this.props.muiTheme.palette.accent2Color : this.props.muiTheme.palette.accent1Color
-      ) :  this.props.muiTheme.palette.primary2Color,
-    }
+    const renderCommentsButton = () => {
+      const read = post.lastVisitedAt;
+      const newComments = post.lastVisitedAt < post.lastCommentedAt;
 
-    const commentCountBadgeStyle = {
-      top: '12px',
-      right: '9px',
-      pointerEvents: 'none',
-      backgroundColor: 'transparent',
-      color: read ? 'white' : 'rgba(0,0,0,0.6)',
-      fontFamily: "freight-sans-pro",
-      zIndex: 2,
-      fontVariantNumeric: "lining-nums",
-      MozFontFeatureSettings: "lnum",
-      WebkitFontFeatureSettings: "lnum",
-      fontFeatureSettings: "lnum",
-    }
+      const commentCountIconStyle = {
+        width:"30px",
+        height:"30px",
+        color: (read && newComments) ? this.props.muiTheme.palette.accent1Color : "rgba(0,0,0,.15)"
```

Comment:
The fontVariants where there for a reason :P 

The text becomes pretty unreadable if you don't do lining-nums.

Reply:
> The text becomes pretty unreadable if you don't do lining-nums.

What does this mean?